File: BEAK_stg2/13913.py
from collections import deque
import sys
MAX = 20000
sys.setrecursionlimit(MAX)

check = [False] * MAX
dist = [-1] * MAX
via =  [-1] * MAX
n,m = mpa(int,input().split())
check[n] = True
dist[n] = 0

q = deque()
q.append(n)

while q:
    now = q.popleft()
    if now-1 >= 0 and not check[now-1]:
        q.append(now-1)
        check(now-1) = True
        dist[now-1] = dist[now] +1
        via[now-1] = now

    if now+1 < MAX and not check[now+1]:
        q.append(now+1)
        check[now+1] = True
        dist[now+1] = dist[now] +1
        via[now+1] = now

    if now*2 < MAX and not check[now*2]:
        q.append(now*2)
        check[now*2]

File: BEAK_yeoun/13913.c
#include <stdio.h>
const int MAX = 200000;
int queue[300000];
int check[200001];
int dist[200001];
int from[200001];
int begin, end;
void push(int x) {
    queue[end++] = x;
}
int pop() {
    return queue[begin++];
}
void print(int n, int m) {
    if (n != m) {
        print(n, from[m]);
    }
    printf("%d ",m);
}
int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    push(n);
    check[n] = 1;
    dist[n] = 0;
    while (begin < end) {
        int now = pop();
        int temp[3] = { now - 1, now + 1, now * 2 };
        for (int k = 0; k < 3; k++) {
            int next = temp[k];
            if (0 <= next && next <= MAX) {
                if (check[next]) continue;
                push(next);
                check[next] = 1;
                dist[next] = dist[now] + 1;
                from[next] = now;
            }
        }
    }
    printf("%d\n", dist[m]);
    print(n, m);
    printf("\n");
    return 0;
}